In [2]:
import pandas as pd
wholefile = pd.read_csv('..//..//..//Documents//Test_sets//Files_for_Disc//testerselectionfiles//All_sound.csv')
ss = pd.read_csv('..//..//..//Documents//Test_sets//Files_for_Disc//testerselectionfiles//Selection.csv')
wholefile.head()


X          Y  SCAMIN  SCAMAX  EXPSOU  NOBJNM  OBJNAM  QUASOU  \
0 -4.153100  50.342776     NaN     NaN     NaN     NaN     NaN       1   
1 -4.162966  50.339084     NaN     NaN     NaN     NaN     NaN       1   
2 -4.163225  50.341281     NaN     NaN     NaN     NaN     NaN       1   
3 -4.166210  50.340903     NaN     NaN     NaN     NaN     NaN       1   
4 -4.167520  50.340732     NaN     NaN     NaN     NaN     NaN       1   

   SOUACC  STATUS         ...           RECDAT  RECIND  SORDAT  \
0     NaN     NaN         ...              NaN     NaN     NaN   
1     NaN     NaN         ...              NaN     NaN     NaN   
2     NaN     NaN         ...              NaN     NaN     NaN   
3     NaN     NaN         ...              NaN     NaN     NaN   
4     NaN     NaN         ...              NaN     NaN     NaN   

                 SORIND  iprown  iprtxt  $orenc  $pcano   Depth  \
0                   NaN     NaN     NaN     NaN     NaN   7.711   
1  GB,GB,surfc,02006193     NaN     NaN     NaN     NaN  10.000   
2                   NaN     NaN     NaN     NaN     NaN   5.751   
3                   NaN     NaN     NaN     NaN     NaN   5.200   
4                   NaN     NaN     NaN     NaN     NaN   5.589   

                   fid  
0  GB 0044236486 00001  
1  GB 0044236488 00001  
2  GB 0044236489 00001  
3  GB 0044236490 00001  
4  GB 0044236491 00001  

[5 rows x 26 columns]

In [3]:
import numpy as np

In [4]:
import matplotlib as p

In [4]:
wholefile.describe()

X            Y  SCAMIN  SCAMAX  EXPSOU  NOBJNM  OBJNAM  \
count  1066.000000  1066.000000     0.0     0.0     0.0     0.0     0.0   
mean     -4.149720    50.352354     NaN     NaN     NaN     NaN     NaN   
std       0.012985     0.008298     NaN     NaN     NaN     NaN     NaN   
min      -4.175090    50.333334     NaN     NaN     NaN     NaN     NaN   
25%      -4.160273    50.346614     NaN     NaN     NaN     NaN     NaN   
50%      -4.149808    50.354418     NaN     NaN     NaN     NaN     NaN   
75%      -4.138527    50.358979     NaN     NaN     NaN     NaN     NaN   
max      -4.124366    50.365973     NaN     NaN     NaN     NaN     NaN   

       QUASOU  SOUACC  STATUS     ...       INFORM  NINFOM  RECDAT  RECIND  \
count  1066.0     0.0     0.0     ...          0.0     0.0     0.0     0.0   
mean      1.0     NaN     NaN     ...          NaN     NaN     NaN     NaN   
std       0.0     NaN     NaN     ...          NaN     NaN     NaN     NaN   
min       1.0     NaN     NaN     ...          NaN     NaN     NaN     NaN   
25%       1.0     NaN     NaN     ...          NaN     NaN     NaN     NaN   
50%       1.0     NaN     NaN     ...          NaN     NaN     NaN     NaN   
75%       1.0     NaN     NaN     ...          NaN     NaN     NaN     NaN   
max       1.0     NaN     NaN     ...          NaN     NaN     NaN     NaN   

       SORDAT  iprown  iprtxt  $orenc  $pcano        Depth  
count     0.0     0.0     0.0     0.0     0.0  1066.000000  
mean      NaN     NaN     NaN     NaN     NaN     9.760483  
std       NaN     NaN     NaN     NaN     NaN     6.949285  
min       NaN     NaN     NaN     NaN     NaN    -3.000000  
25%       NaN     NaN     NaN     NaN     NaN     5.687750  
50%       NaN     NaN     NaN     NaN     NaN     7.423500  
75%       NaN     NaN     NaN     NaN     NaN    10.531250  
max       NaN     NaN     NaN     NaN     NaN    38.986000  

[8 rows x 24 columns]

In [5]:
ss.describe()

X           Y  SCAMIN  SCAMAX  EXPSOU  NOBJNM  OBJNAM  QUASOU  \
count  121.000000  121.000000     0.0     0.0     0.0     0.0     0.0   121.0   
mean    -4.144160   50.358472     NaN     NaN     NaN     NaN     NaN     1.0   
std      0.003022    0.001823     NaN     NaN     NaN     NaN     NaN     0.0   
min     -4.149466   50.355111     NaN     NaN     NaN     NaN     NaN     1.0   
25%     -4.146896   50.357023     NaN     NaN     NaN     NaN     NaN     1.0   
50%     -4.143884   50.358551     NaN     NaN     NaN     NaN     NaN     1.0   
75%     -4.141680   50.360011     NaN     NaN     NaN     NaN     NaN     1.0   
max     -4.139071   50.361506     NaN     NaN     NaN     NaN     NaN     1.0   

       SOUACC  STATUS     ...      NINFOM  RECDAT  RECIND  SORDAT  SORIND  \
count     0.0     0.0     ...         0.0     0.0     0.0     0.0     0.0   
mean      NaN     NaN     ...         NaN     NaN     NaN     NaN     NaN   
std       NaN     NaN     ...         NaN     NaN     NaN     NaN     NaN   
min       NaN     NaN     ...         NaN     NaN     NaN     NaN     NaN   
25%       NaN     NaN     ...         NaN     NaN     NaN     NaN     NaN   
50%       NaN     NaN     ...         NaN     NaN     NaN     NaN     NaN   
75%       NaN     NaN     ...         NaN     NaN     NaN     NaN     NaN   
max       NaN     NaN     ...         NaN     NaN     NaN     NaN     NaN   

       iprown  iprtxt  $orenc  $pcano       Depth  
count     0.0     0.0     0.0     0.0  121.000000  
mean      NaN     NaN     NaN     NaN   11.866512  
std       NaN     NaN     NaN     NaN    7.299043  
min       NaN     NaN     NaN     NaN    2.904000  
25%       NaN     NaN     NaN     NaN    6.494000  
50%       NaN     NaN     NaN     NaN    9.271000  
75%       NaN     NaN     NaN     NaN   16.338000  
max       NaN     NaN     NaN     NaN   33.233000  

[8 rows x 25 columns]

#Know from above that we have 1066 soundings and a selection of 121

#Need to remove columns which are not useful

In [6]:
wholefile_d= wholefile.loc[:,['X','Y','Depth','fid']]
wholefile_d.head(3)

X          Y   Depth                  fid
0 -4.153100  50.342776   7.711  GB 0044236486 00001
1 -4.162966  50.339084  10.000  GB 0044236488 00001
2 -4.163225  50.341281   5.751  GB 0044236489 00001

#old method

#wholefile_c= wholefile.dropna(axis = 'columns',how='any')

#wholefile_c.describe()

#dropped all columns with NaN values, know from describe above that all soundings are QUASOU 1 so this can be removed too

#wholefile_d = wholefile_c.drop('QUASOU', axis =1)
------


#need to add a column to say if the sounding has been selected 

In [7]:
ss2 = ss.dropna(axis = 'columns',how='any')
ss2 = ss.loc[:,['X','Y','Depth']]

In [8]:
#ss2['Selection'] = 1
#ss2.columns

In [9]:
#wholefile_e = pd.merge(wholefile_d,ss2,how = 'outer',on =['X','Y'])
#wholefile_e.head(10)
# checked depth values the same wholefile_e[wholefile_e.Selection == 1]
# Now need to make unselected Selection = 0 and and delete Depth_y

In [10]:
#wholefile_f = wholefile_e.drop('Depth_y', axis =1)
#wholefile_f.describe()

In [11]:
#replace NaN

#final = wholefile_f.fillna(value = 0)
#final.head(5)

#final.describe()

geopandas - buffer to a tollerance
density of data?


In [12]:
wholefile_d['Selection'] = 0
wholefile_d.head(4)


X          Y   Depth                  fid  Selection
0 -4.153100  50.342776   7.711  GB 0044236486 00001          0
1 -4.162966  50.339084  10.000  GB 0044236488 00001          0
2 -4.163225  50.341281   5.751  GB 0044236489 00001          0
3 -4.166210  50.340903   5.200  GB 0044236490 00001          0

In [13]:
ss2.head(4)

X          Y   Depth
0 -4.148672  50.361437  13.668
1 -4.146557  50.361368  12.973
2 -4.145353  50.361506  11.248
3 -4.143561  50.361486  12.456

In [14]:
min_pos = []
errors = []
for i in range(0,len(ss2)):
    x1 = ss2.iloc[i][0]
    y1 = ss2.iloc[i][1]
    #print(x,y)
    
    dist_vect = []
    
    for j in range(0,len(wholefile_d)):
        x2= wholefile_d.iloc[j][0]
        y2= wholefile_d.iloc[j][1]
        
        #distance between x and ys
        dist = np.sqrt(np.abs((x2-x1)+(y2-y1)))
        dist_vect.append(dist)
    
    min_value = dist_vect.index(min(dist_vect))
    min_pos.append(min_value)
    errors.append(dist_vect[min_value])
#print(min_pos)
#print(errors)        
        

In [15]:
len(errors)

121

In [16]:
wholefile_d.iloc[min_pos,4] = 1
wholefile_d.head(300)

X          Y   Depth                  fid  Selection
0   -4.153100  50.342776   7.711  GB 0044236486 00001          0
1   -4.162966  50.339084  10.000  GB 0044236488 00001          0
2   -4.163225  50.341281   5.751  GB 0044236489 00001          0
3   -4.166210  50.340903   5.200  GB 0044236490 00001          0
4   -4.167520  50.340732   5.589  GB 0044236491 00001          0
5   -4.153763  50.341100  10.466  GB 0044236492 00001          0
6   -4.165542  50.340207   4.969  GB 0044236493 00001          0
7   -4.166643  50.340196   5.361  GB 0044236494 00001          0
8   -4.163679  50.340719   5.470  GB 0044236495 00001          0
9   -4.160701  50.340282   9.870  GB 0044236496 00001          0
10  -4.161994  50.339834   8.298  GB 0044236497 00001          0
11  -4.157841  50.340599  10.715  GB 0044236498 00001          0
12  -4.155901  50.340693  11.946  GB 0044236499 00001          0
13  -4.159856  50.339626  10.923  GB 0044236500 00001          0
14  -4.163979  50.338866   9.550  GB 0044236501 00001          0
15  -4.164351  50.339391   7.652  GB 0044236502 00001          0
16  -4.162625  50.338591  10.623  GB 0044236503 00001          0
17  -4.165319  50.340752   5.492  GB 0044236504 00001          0
18  -4.159681  50.340891   9.785  GB 0044236505 00001          0
19  -4.165977  50.339579   7.580  GB 0044236506 00001          0
20  -4.152730  50.344317   7.544  GB 0044236507 00001          0
21  -4.153675  50.345899   6.358  GB 0044236508 00001          0
22  -4.151428  50.346918   6.482  GB 0044236509 00001          0
23  -4.152392  50.346483   6.236  GB 0044236510 00001          0
24  -4.152940  50.351148   3.288  GB 0044236511 00001          0
25  -4.151641  50.349119   6.988  GB 0044236512 00001          0
26  -4.152215  50.350298   5.534  GB 0044236513 00001          0
27  -4.153211  50.348982   7.319  GB 0044236514 00001          0
28  -4.153665  50.341366  10.882  GB 0044236515 00001          0
29  -4.151912  50.350745   5.000  GB 0044236516 00001          0
..        ...        ...     ...                  ...        ...
270 -4.132115  50.356068   6.099  GB 0044236758 00001          0
271 -4.133024  50.355596   6.323  GB 0044236759 00001          0
272 -4.129447  50.355725   4.427  GB 0044236760 00001          0
273 -4.129884  50.355701   5.578  GB 0044236761 00001          0
274 -4.131013  50.356067   5.701  GB 0044236762 00001          0
275 -4.138496  50.355663  11.184  GB 0044236763 00001          0
276 -4.139498  50.356313  10.703  GB 0044236764 00001          1
277 -4.133924  50.357649   5.762  GB 0044236765 00001          0
278 -4.139507  50.357517  16.916  GB 0044236766 00001          1
279 -4.139071  50.355662   8.925  GB 0044236767 00001          1
280 -4.139347  50.354611   8.965  GB 0044236768 00001          0
281 -4.139526  50.353526   8.460  GB 0044236769 00001          0
282 -4.139859  50.358179  22.111  GB 0044236770 00001          1
283 -4.170859  50.360993   8.770  GB 0044236771 00001          0
284 -4.169905  50.363073  26.852  GB 0044236772 00001          0
285 -4.170051  50.356481  25.180  GB 0044236773 00001          0
286 -4.171976  50.364080  27.937  GB 0044236774 00001          0
287 -4.172220  50.361614   8.292  GB 0044236775 00001          0
288 -4.172174  50.361228   7.938  GB 0044236776 00001          0
289 -4.171850  50.360376   3.970  GB 0044236777 00001          0
290 -4.172194  50.360582   3.420  GB 0044236778 00001          0
291 -4.171650  50.360060   7.311  GB 0044236779 00001          0
292 -4.171575  50.359646   3.020  GB 0044236780 00001          0
293 -4.170224  50.361516  25.475  GB 0044236781 00001          0
294 -4.169932  50.362282  28.926  GB 0044236782 00001          0
295 -4.170319  50.360143  12.343  GB 0044236783 00001          0
296 -4.170658  50.359674  18.036  GB 0044236784 00001          0
297 -4.169976  50.360633  24.369  GB 0044236785 00001          0
298 -4.170920  50.359985   8.639  GB 0044236786 00001          0
299 -4.171137  50.360488   8.587  GB 0044236787

# Simple merge

In [17]:
wholefile_d.to_csv('..//..//..//Documents//Test_sets//Files_for_Disc//testerselectionfiles//Selection_calc.csv')

In [23]:
survey = pd.read_csv('..//..//..//Documents//Test_sets//Testing_Sample//HI1469//HI1469.txt')
ss = pd.read_csv('..//..//..//Documents//Test_sets//Testing_Sample//HI1469//All_sound.csv')

survey.describe()


Y             X         Depth
count  1.908734e+06  1.908734e+06  1.908734e+06
mean   5.034771e+01 -4.144780e+00  9.636321e+00
std    8.394234e-03  1.098222e-02  4.698753e+00
min    5.033292e+01 -4.168793e+00 -1.065000e+00
25%    5.034067e+01 -4.153674e+00  6.918000e+00
50%    5.034716e+01 -4.144443e+00  8.974000e+00
75%    5.035465e+01 -4.136058e+00  1.081700e+01
max    5.036627e+01 -4.113887e+00  4.029800e+01

In [16]:
#A_1= survey.iloc[:,[1,0,2]]
#A_1.head()

In [17]:
#A_1.columns = ['X', 'Y', 'Depth']
#A_1.describe()

In [18]:
ss.head(3)

X          Y  SCAMIN  SCAMAX  EXPSOU  NOBJNM  OBJNAM  QUASOU  \
0 -4.153100  50.342776     NaN     NaN     NaN     NaN     NaN       1   
1 -4.162966  50.339084     NaN     NaN     NaN     NaN     NaN       1   
2 -4.163225  50.341281     NaN     NaN     NaN     NaN     NaN       1   

   SOUACC  STATUS         ...           RECDAT  RECIND  SORDAT  \
0     NaN     NaN         ...              NaN     NaN     NaN   
1     NaN     NaN         ...              NaN     NaN     NaN   
2     NaN     NaN         ...              NaN     NaN     NaN   

                 SORIND  iprown  iprtxt  $orenc  $pcano   Depth  \
0                   NaN     NaN     NaN     NaN     NaN   7.711   
1  GB,GB,surfc,02006193     NaN     NaN     NaN     NaN  10.000   
2                   NaN     NaN     NaN     NaN     NaN   5.751   

                   fid  
0  GB 0044236486 00001  
1  GB 0044236488 00001  
2  GB 0044236489 00001  

[3 rows x 26 columns]

In [19]:
#A_1= survey.loc[:,['Lon','Lat','Depth']]
B_1= ss.loc[:,['X','Y']]


In [20]:
B_1['Selection'] = 1
B_1.describe()

X            Y  Selection
count  1066.000000  1066.000000     1066.0
mean     -4.149720    50.352354        1.0
std       0.012985     0.008298        0.0
min      -4.175090    50.333334        1.0
25%      -4.160273    50.346614        1.0
50%      -4.149808    50.354418        1.0
75%      -4.138527    50.358979        1.0
max      -4.124366    50.365973        1.0

In [24]:
A_2 = pd.merge(survey,B_1,how = 'outer',on =['X','Y'])
A_2.head(10)

Y         X   Depth  Selection
0  50.337858 -4.164087  11.936        NaN
1  50.337870 -4.164108  11.963        NaN
2  50.337868 -4.164094  11.943        NaN
3  50.337894 -4.164136  11.978        NaN
4  50.337895 -4.164130  11.960        NaN
5  50.337884 -4.164084  11.965        NaN
6  50.337915 -4.164167  11.950        NaN
7  50.337917 -4.164139  11.923        NaN
8  50.337910 -4.164126  11.929        NaN
9  50.337909 -4.164103  11.947        NaN

In [25]:
A_2.describe()

Y             X         Depth  Selection
count  1.908891e+06  1.908891e+06  1.908734e+06     1066.0
mean   5.034771e+01 -4.144782e+00  9.636321e+00        1.0
std    8.394564e-03  1.098380e-02  4.698753e+00        0.0
min    5.033292e+01 -4.175090e+00 -1.065000e+00        1.0
25%    5.034067e+01 -4.153676e+00  6.918000e+00        1.0
50%    5.034716e+01 -4.144444e+00  8.974000e+00        1.0
75%    5.035465e+01 -4.136059e+00  1.081700e+01        1.0
max    5.036627e+01 -4.113887e+00  4.029800e+01        1.0

In [26]:
#A_3 = A_2.drop('Depth_y', axis =1)

#A_3.describe()

In [27]:
final = A_2.fillna(value = 0)
final.head(5)

Y         X   Depth  Selection
0  50.337858 -4.164087  11.936        0.0
1  50.337870 -4.164108  11.963        0.0
2  50.337868 -4.164094  11.943        0.0
3  50.337894 -4.164136  11.978        0.0
4  50.337895 -4.164130  11.960        0.0

In [28]:
final.to_csv('..//..//..//Documents//Test_sets//Testing_Sample//HI1469//Edits//HI1469_C_MERGE.csv')
    